tests/phpunit/Git/CommandLineGitTest.php

```diff
@@ -56,17 +57,26 @@
 #[CoversClass(CommandLineGit::class)]
 final class CommandLineGitTest extends TestCase
 {
+    private ShellCommandLineExecutor&MockObject $commandLineMock;
+
+    private Git $git;
+
+    protected function setUp(): void
+    {
+        $this->commandLineMock = $this->createMock(ShellCommandLineExecutor::class);
+
+        $this->git = new CommandLineGit($this->commandLineMock);
+    }
```

Comment:
I'm pretty sure that as a best practices it was suggested (in one of the Sebastian's talk) to create such services in a private method and reuse in each test method.

What was the reason to move it to `setUp` and in which way is it better?